src/Microsoft.DotNet.Cli.Utils/Command.cs

```diff
@@ -85,6 +85,11 @@ private Command(CommandSpec commandSpec)
 
             return command;
         }
+
+        public static Command Create(CommandSpec commandSpec)
```

Comment:
Fair enough, the only reason I needed to expose this was so that the `ProjectDependenciesCommandFactory` that we now use in the test runner needed to be able to call into the resolvers itself rather than going through the Command.Create facade. We could've have added infinitely many `Create` functions, but it seems to overlap with the responsibility of the CommandResolvers.

IMO the right thing moving forward will be to move things to the explicit factories (e.g. ProjectDependenciesCommandFactory, DotnetCommandFactory, CommandFactory) and move away from Command.Create. The flexibility we need seems to be better served through separate implementations.

What do you think of that approach?